### ISSUE TITLE
Nested TOC broken (flattened) in 0.4.2

### ISSUE THREAD

**Issue by flodolo** [badboy/mdbook-toc#3]
Locally I had 0.3.0 installed, and the TOC for this page was generated correctly
* [Markdown Source](https://github.com/mozilla-l10n/documentation/blob/master/src/products/firefox_desktop/firefox_l10n_faqs.md)
* [Generated HTML](https://mozilla-l10n.github.io/documentation/products/firefox_desktop/firefox_l10n_faqs.html) using mdbook-toc 0.4.2, mdbook 0.3.7

As you can see, the TOC is completely flat.

**Comment by flodolo** [comment]
Note: I assume it broke with 0.4.2, I think I would have noticed the issue before.

I see [this commit](https://github.com/badboy/mdbook-toc/commit/b220318021e7d821be02a210b2f7fda865d94d9d), but the structure seems correct to me (no holes)

```
# Firefox L10N FAQs
## General
### Who can I contact if I have more questions?
### Should I enable my feature only for English, or ship it in English for everyone?
## Schedule and release cycle
### I landed strings in mozilla-central, when are they going to be localized?
### Can I uplift a patch to Beta or Release?
### I need to add new strings for version X, do I still have time?
## Development
### Where can I find guidelines for developers?
### How can I get my strings localized?
### Where can I find the localized strings?
### Can I land content without exposing it for localization?
### How can I test a different locale?
## Translation completion and community
### Is my feature going to be localized in language X before launch?
### How do I communicate with localizers about my feature or patch?
### I see pending suggestions in Pontoon, how can I get them approved?
### How can I check if my feature is localized?
### How can I see all translations available for a specific string?
```

**Comment by flodolo** [comment]
To close the gap, 0.4.1 works as expected locally.

<img width="776" alt="Schermata 2020-06-09 alle 14 53 57" src="https://user-images.githubusercontent.com/3644868/84149636-12ab7e80-aa61-11ea-9ef9-670fcdbd77bd.png">

**Comment by badboy** [comment]
Fixed in [commit 2d5b4021](https://github.com/badboy/mdbook-toc/commit/2d5b40214c37bca75f57e5583f224d67fc452dcf) and released in [v0.4.3](https://github.com/badboy/mdbook-toc/releases/tag/0.4.3)
